### PR TITLE
Display storage dump with state before revert

### LINKED ISSUE #92: storage dump is wrong after reversion
storage dump will pick up the state after the state inside the computation already got rolled back. to properly fix this, we need to patch py-evm to save the state inside of `apply_computation` / `apply_message` before the state gets reverted (although, that would be a performance hit).

### PR DESCRIPTION
**Pull request by DanielSchiavini**
### What I did
- Save the contract `repr` with the computation in case of an error
- Use this value to show the storage state when the computation errored
- This should fix #92

### How I did it
- Investigated how py-evm handles the snapshots
  - However, the data gets completely deleted after revert. To save all the [deleted data](https://github.com/ethereum/py-evm/blob/d625109bd92ab9a811c9ab64d375448b3fe7aecf/eth/db/account.py#L438-L444) would be quite cumbersome
- Instead of saving the whole journal, it makes more sense to only save the data we need when we need it
- Therefore I only extract the `repr(contract)` before the revert, as this is the information used
- Alternatively we could extract more data at this point (e.g. `storage.dump()`)
- The implementation is in the `environment` as this is specific to py-evm

### How to verify it
- Test is included; Please advise on extra test/corner cases that I did not identify.

### Description for the changelog
- Display the storage dump with the state before the computation was reverted

### Cute Animal Picture
image

### CODE REVIEW

File: boa/environment.py [hunk 1]
```diff
@@ -344,7 +345,12 @@ def apply_computation(cls, state, msg, tx_ctx):
         # print("ENTER", Address(msg.code_address or bytes([0]*20)), contract)
         if contract is None or not cls.env._fast_mode_enabled:
             # print("SLOW MODE")
-            return super().apply_computation(state, msg, tx_ctx)
+            computation = super().apply_computation(state, msg, tx_ctx)
+            if computation.is_error:
+                # After the computation is applied with an error the state is reverted
+                # Before the revert, save the contract representation for the error message
+                computation._contract_repr_before_revert = repr(contract)
```
Comment (DanielSchiavini) [inline boa/environment.py:352]: Alternatively we could extract more data at this point (e.g. storage.dump())

**Review by charles-cooper — COMMENTED** [review]
let's roll back the changes to the CI for now

File: boa/environment.py (hunk 1 above)
Comment (charles-cooper) [reply boa/environment.py:352]: let's pull the whole storage dump for now. it seems like it could come in handy

Comment (charles-cooper) [reply boa/environment.py]: never mind -- i agree with your approach of getting the whole contract repr
